control-plane/config-entries/controllers/configentry_controller.go

```diff
@@ -125,7 +130,9 @@ func (r *ConfigEntryController) ReconcileEntry(ctx context.Context, crdCtrl Cont
 		// then let's add the finalizer and update the object. This is equivalent
 		// registering our finalizer.
 		if !containsString(configEntry.GetFinalizers(), FinalizerName) {
-			configEntry.AddFinalizer(FinalizerName)
+			addPatch := crdCtrl.AddFinalizersPatch(configEntry, FinalizerName)
+			crdCtrl.Patch(ctx, configEntry, addPatch)
```

Comment:
Is there supposed to be error handling here? I'm on the fence about this, but do you think we should maybe do the `Patch()` call inside of `AddFinalizerPatch()` ?

Reply:
Yes, just added the error handling, thank you for catching!

Definitely correct me if I'm wrong, but I don't think there's a way to do that without having an implementation of AddFinalizerPatch() for each controller. Since AddFinalizerPatch is a function on the FinalizerPatcher which is embedded in each controller right now, it doesn't have access to the Patch() function on the client. 

If we choose to un-embed the FinalizerPatcher and implement the AddFinalizersPatch for each controller, we could do the patch inside of AddFinalizerPatch. Do you have a preference? I don't have a strong one, just chose this for less code.

Reply:
This seems fine. Changing it would shuffle complexity into other locations, so we can leave this as-is.